### PR TITLE
fix: savings plugin

### PR DESCRIPTION
**Pull request by yanzhihong23**
## Description

Closes [MF-86](https://mask.atlassian.net/browse/MF-86)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx [hunk 1]
```diff
@@ -21,7 +21,8 @@ import {
     useWeb3,
     ZERO_ADDRESS,
 } from '@masknet/web3-shared-evm'
-import { FormattedCurrency, LoadingAnimation, TokenAmountPanel, TokenIcon } from '@masknet/shared'
+import { FormattedCurrency, LoadingAnimation, TokenIcon } from '@masknet/shared'
+import { TokenAmountPanel } from '../../../web3/UI/TokenAmountPanel'
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:22]: Those two `<TokenAmountPanel />` are almost the same thing. Could you please keep one and remove the other.

Comment (hanszhoou) [reply packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:22]: https://github.com/DimensionDev/Maskbook/commits/develop/packages/mask/src/web3/UI/TokenAmountPanel.tsx
https://github.com/DimensionDev/Maskbook/commits/develop/packages/shared/src/UI/components/TokenAmountPanel/index.tsx

FYI @yanzhihong23

**Comment by guanbinrui** [comment]
Any updates?

File: packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx [hunk 2]
```diff
@@ -198,7 +199,8 @@ export function SavingsForm({ chainId, protocol, tab, onClose }: SavingsFormProp
                         })
                     }))
                 ) {
-                    throw new Error('Failed to deposit token.')
+                    // eslint-disable-next-line
+                    return Promise.reject(Promise.reject('Failed to deposit token.'))
```
Comment (UncleBill) [inline packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:184]: Why call `Promise.reject` twice?
```suggestion
                    return Promise.reject(new Error('Failed to deposit token.'))
```

Comment (yanzhihong23) [reply packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:184]: > 

This is a hack for useAsyncFn reject.

Comment (Jack-Works) [reply packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx:184]: Can you explain what hack is it? I don't get it.

**Pull request merged**